core/src/test/java/brooklyn/location/basic/SshMachineLocationIntegrationTest.java

```diff
@@ -75,7 +81,36 @@ public void testExtractingConnectablePassphraselessKey() throws Exception {
         ByteArrayOutputStream out = new ByteArrayOutputStream();
         int result = tool.execCommands(MutableMap.<String,Object>of("out", out), Arrays.asList("date"));
         Assert.assertTrue(out.toString().contains(" 20"), "out="+out);
-        Assert.assertEquals(result, 0);
+        assertEquals(result, 0);
+    }
+
+    @Test(groups = "Integration")
+    public void testExecScriptScriptDirFlagIsRespected() throws Exception {
+        // For explanation of (some of) the magic behind this command, see http://stackoverflow.com/a/229606/68898
+        final String command = "if [[ \"$0\" == \"/var/tmp/\"* ]]; then true; else false; fi";
+
+        LocalhostMachineProvisioningLocation lhp = (LocalhostMachineProvisioningLocation) mgmt.getLocationRegistry().resolve("localhost", true, null).orNull();
+        SshMachineLocation sm = lhp.obtain();
+
+        Map<String, Object> props = ImmutableMap.<String, Object>builder()
+                .put(SshTool.PROP_SCRIPT_DIR.getName(), "/var/tmp")
+                .build();
+        sm.execScript(props, "Test script directory execution", ImmutableList.of(command));
     }
 
+    @Test(groups = "Integration")
+    public void testLocationScriptDirConfigIsRespected() throws Exception {
+        // For explanation of (some of) the magic behind this command, see http://stackoverflow.com/a/229606/68898
+        final String command = "if [[ \"$0\" == \"/var/tmp/\"* ]]; then true; else false; fi";
+
+        Map<String, Object> locationConfig = ImmutableMap.<String, Object>builder()
+                .put(SshMachineLocation.SCRIPT_DIR.getName(), "/var/tmp")
+                .build();
+
+        LocalhostMachineProvisioningLocation lhp = (LocalhostMachineProvisioningLocation) mgmt.getLocationRegistry().resolve("localhost", true, locationConfig).orNull();
```

Comment:
The default for the `managed` param is true, so you can just miss that out I think.